src/users/register.js

```diff
@@ -6,10 +6,30 @@ import {Navigate} from "react-router";
 
 const Register = () => {
     const {currentUser} = useSelector((state) => state.users)
-    const [username, setUsername] = useState('alice')
-    const [password, setPassword] = useState('alice1234')
+    const [username, setUsername] = useState("")
+    const [password, setPassword] = useState("")
+    const [validatePassword, setValidatePassword] = useState("")
+    const [role, setRole] = useState(false)
+    const [error, setError] = useState(null)
     const dispatch = useDispatch()
     const handleRegisterBtn = () => {
+        if (!username || !password || !validatePassword) {
+            setError('All fields should be filled out')
+            return
+        }
+        if (username.length < 2 || username.length > 20) {
+            setError('Username should be between 2 and 20 characters')
+            return
+        }
+        if (password.length < 2 || password.length > 20) {
+            setError('Password should be between 2 and 20 characters')
+            return
+        }
+        if (password !== validatePassword) {
+            setError('Passwords must match')
+            return
+        }
+        setError(null)
         dispatch(registerThunk({username, password}))
```

Comment:
If the checkbox is working correctly then you can modify the registerThunk and any related functions to pass in the role as well

```
admin ? dispatch(registerThunk({username, password, 'ADMIN'}) : dispatch(registerThunk({username, password, 'USER'})
```